Sources/Build/llbuild.swift

```diff
@@ -0,0 +1,183 @@
+        stream <<< "targets:\n"
+        for target in [targets.test, targets.main] {
+            stream <<< "  " <<< Format.asJSON(target.name) <<< ": " <<< Format.asJSON(target.cmds.flatMap{$0.tool.outputs}) <<< "\n"
+        }
+        stream <<< "default: " <<< Format.asJSON(targets.main.name) <<< "\n"
+        stream <<< "commands: \n"
+        for command in targets.allCommands {
+            stream <<< "  " <<< Format.asJSON(command.name) <<< ":\n"
+            command.tool.append(to: stream)
+            stream <<< "\n"
+        }
+        try localFileSystem.writeFileContents(path, bytes: stream.bytes)
+    }
+
+    /// Create link command for products.
+    private func createLinkCommand(_ buildProduct: ProductBuildDescription) -> Command {
+        let tool: ToolProtocol
+        // Create archive tool for static library and shell tool for rest of the products.
+        if buildProduct.product.type == .Library(.Static) {
+            tool = ArchiveTool(inputs: buildProduct.objects.map{$0.asString}, outputs: [buildProduct.binary.asString])
+        } else {
+            tool = ShellTool(
+                description: "Linking \(buildProduct.binary.prettyPath)",
+                inputs: buildProduct.objects.map{$0.asString},
+                outputs: [buildProduct.binary.asString],
+                args: buildProduct.linkArguments())
+        }
+        return Command(name: buildProduct.product.targetName, tool: tool)
+    }
+
+    /// Create command for Swift target description.
+    private func createSwiftCommand(_ target: SwiftTargetDescription) -> Command {
+        // Compute inputs.
+        var inputs = target.module.sources.paths.map{ $0.asString }
+        for dependency in target.module.dependencies {
+            // Ignore C Modules.
+            if dependency is CModule { continue }
+            switch plan.targetMap[dependency] {
+            case .swift(let target)?:
+                inputs += [target.moduleOutputPath.asString]
+            case .clang(let target)?:
+                inputs += target.objects.map{$0.asString}
+            case nil:
+                fatalError("unexpected: module \(dependency) not in target map \(plan.targetMap)")
+            }
+        }
+
+        let tool = SwiftCompilerTool(target: target, inputs: inputs)
+        return Command(name: target.module.targetName, tool: tool)
+    }
+
+    /// Create commands for Clang targets.
+    private func createClangCommands(_ target: ClangTargetDescription) -> [Command] {
+        return target.compilePaths().map { path in
+            var args = target.basicArguments()
+            args += ["-MD", "-MT", "dependencies", "-MF", path.deps.asString]
+            args += ["-c", path.source.asString, "-o", path.object.asString]
+            let clang = ClangTool(desc: "Compile \(target.module.name) \(path.filename.asString)",
+                                  //FIXME: Should we add build time dependency on dependent modules?
```

Comment:
Yes, it would need to depend on anything that could generate headers.  For example, I think there is an option to the Swift compiler to generate a bridging header to expose the Swift API to C and ObjC, and this would need to wait for that header to be created (I don't know if SwiftPM is explicitly stated as not supporting C dependencies on Swift content, but if that's supposed to be supported, then there would need to be a dependency here).